cmd/kubeadm/app/phases/kubelet/flags_unix.go

```diff
@@ -0,0 +1,89 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/klog"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/features"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+)
+
+// buildKubeletArgMap takes a kubeletFlagsOpts object and builds based on that a string-string map with flags
+// that should be given to the local kubelet daemon.
+func buildKubeletArgMap(opts kubeletFlagsOpts) map[string]string {
+	kubeletFlags := map[string]string{}
+
+	if opts.nodeRegOpts.CRISocket == constants.DefaultDockerCRISocket {
+		// These flags should only be set when running docker
+		kubeletFlags["network-plugin"] = "cni"
+		driver, err := kubeadmutil.GetCgroupDriverDocker(opts.execer)
+		if err != nil {
+			klog.Warningf("cannot automatically assign a '--cgroup-driver' value when starting the Kubelet: %v\n", err)
+		} else {
+			kubeletFlags["cgroup-driver"] = driver
+		}
+		if opts.pauseImage != "" {
+			kubeletFlags["pod-infra-container-image"] = opts.pauseImage
+		}
+	} else {
+		kubeletFlags["container-runtime"] = "remote"
+		kubeletFlags["container-runtime-endpoint"] = opts.nodeRegOpts.CRISocket
+	}
+
+	if opts.registerTaintsUsingFlags && opts.nodeRegOpts.Taints != nil && len(opts.nodeRegOpts.Taints) > 0 {
+		taintStrs := []string{}
+		for _, taint := range opts.nodeRegOpts.Taints {
+			taintStrs = append(taintStrs, taint.ToString())
+		}
+
+		kubeletFlags["register-with-taints"] = strings.Join(taintStrs, ",")
+	}
+
+	ok, err := opts.isServiceActiveFunc("systemd-resolved")
+	if err != nil {
+		klog.Warningf("cannot determine if systemd-resolved is active: %v\n", err)
+	}
+	if ok {
+		kubeletFlags["resolv-conf"] = "/run/systemd/resolve/resolv.conf"
+	}
+
+	// Pass the "--hostname-override" flag to the kubelet only if it's different from the hostname
+	nodeName, hostname, err := GetNodeNameAndHostname(opts.nodeRegOpts)
+	if err != nil {
+		klog.Warning(err)
+	}
+	if nodeName != hostname {
+		klog.V(1).Infof("setting kubelet hostname-override to %q", nodeName)
+		kubeletFlags["hostname-override"] = nodeName
+	}
+
+	// TODO: Conditionally set `--cgroup-driver` to either `systemd` or `cgroupfs` for CRI other than Docker
+
+	// TODO: The following code should be removed after dual-stack is GA.
+	// Note: The user still retains the ability to explicitly set feature-gates and that value will overwrite this base value.
+	if enabled, present := opts.featureGates[features.IPv6DualStack]; present {
```

Comment:
do you happen to know what is the state of ipv6 and dual-stack currently on Windows and if it is disabled, what are the plans for the near future?

i think this feature gate can be passed on Windows too, even if it's a NO-OP.

Reply:
There was someone from MS that spoke about working on dual-stack support in our last sig call. Since that seems to be on the horizon, I'll pass this on to Windows as well.

Reply:
ok :+1: